src/app/mesh/qgsmaptooleditmeshframe.cpp

```diff
@@ -1926,6 +1928,32 @@ void QgsMapToolEditMeshFrame::reindexMesh()
   mCurrentLayer->reindex( transform, true );
 }
 
+void QgsMapToolEditMeshFrame::onUndoRedo()
+{
+  switch ( mCurrentState )
+  {
+    case Digitizing:
+      break;
+    case AddingNewFace:
+      mNewFaceBand->reset( Qgis::GeometryType::Polygon );
+      mNewFaceCandidate.clear();
+      mNewVerticesForNewFaceCandidate.clear();
+      mCurrentState = Digitizing;
+      break;
+    case MovingSelection:
+      mCurrentState = Digitizing;
+      mMovingEdgesRubberband->reset( Qgis::GeometryType::Line );
+      mMovingFacesRubberband->reset( Qgis::GeometryType::Polygon );
+      mMovingFreeVertexRubberband->reset( Qgis::GeometryType::Point );
+      mCadDockWidget->setEnabledZ( mCadDockWidget->cadEnabled() );
+      break;
```

Comment:
If advanced digitizing is enabled, the last cad dock widget point remains on the canvas.
Calling `mCadDockWidget->clear()` in those two cases would probably fix that too.